bin/pkg.ml

```diff
@@ -438,22 +390,15 @@ module Lock = struct
         & flag
         & info [ "all-contexts" ] ~doc:"Generate the lockdir for all contexts")
     and+ version_preference = Version_preference.term
-    and+ use_env_from_current_system =
+    and+ dont_poll_system_solver_variables =
       Arg.(
         value
         & flag
         & info
-            [ "use-env-from-current-system" ]
+            [ "dont-poll-system-solver-variables" ]
             ~doc:
-              "Set opam system environment variables based on the current system when \
-               solving dependencies. This will restrict packages to just those which can \
-               be installed on the current system. Note that this will mean that the \
-               generated lockdir may not be compatible with other systems. If the build \
-               context(s) specify system environment variables then the solver will use \
-               the union of the environment variables inferred from the current system \
-               and the environment variables set in the build context(s). If the current \
-               system and the build context(s) disagree on the value of an environment \
-               variable then an error is raised.")
+              "Don't derive system solver variables from the current system. Values \
```

Comment:
Could you explain the option a bit more?

What happens when the solver needs to evaluate `%{os}` and we passed this option.

Reply:
I've updated the description. Hopefully it is clearer now.